### PR TITLE
Integration of PartialINDs and fixing of evaluations

### PR DESCRIPTION
**Pull request by dkuska**

### CODE REVIEW

File: pysrc/models/ind.py [hunk 1]
```diff
@@ -11,7 +11,7 @@ class IND:
 
     referenced: list[ColumnInformation]
 
-    errors: list[ErrorMetric] = field(default_factory=list, compare=False)
+    errors: list[MissingValues] = field(default_factory=list, compare=False) #NOTE: THIS IS JUST TEMPORARY SO THAT WE CAN ACTUALLY WORK WITH PARTIAL SPIDER!!!
```
Comment (ClFeSc) [inline pysrc/models/ind.py]: I'll try and look into it.

File: pysrc/models/metanome_run.py [hunk 2]
```diff
@@ -34,8 +35,10 @@ class MetanomeRunConfiguration:
     is_baseline: bool
 
     def __hash__(self) -> int:
-        return hash((self.arity, tuple(self.sampling_rates), tuple(self.sampling_methods), self.time, self.source_dir, tuple(self.source_files), self.tmp_folder, self.results_folder,
-        self.result_suffix, self.output_folder, self.output_file, self.clip_output, self.header, self.print_inds, self.create_plots, self.is_baseline))
+        return hash((self.arity, tuple(self.sampling_rates), tuple(self.sampling_methods), self.time, self.source_dir,
```
Comment (ClFeSc) [inline pysrc/models/metanome_run.py]: I think you should add `algorithm` to this `hash`.

File: pysrc/models/metanome_run.py [hunk 3]
```diff
@@ -152,6 +158,7 @@ def parse_results(result_file_name: str, arity: str, results_folder: str, print_
 
     for line in lines:
         line_json = json.loads(line)
+        error: list[ErrorMetric] = []
```
Comment (ClFeSc) [inline pysrc/models/metanome_run.py]:
```suggestion
        errors: list[ErrorMetric] = []
```

File: pysrc/models/metanome_run.py [hunk 4]
```diff
@@ -351,22 +383,39 @@ def run_as_compared_csv_line(run: MetanomeRun, baseline: MetanomeRunResults) ->
...
         else:
-            fname, sampling_rate, sampling_method  = sampled_file, '1.0', 'None'
+            fname, sampling_rate, sampling_method = sampled_file, '1.0', 'None'
 
         file_names.append(fname)
         methods.append(sampling_method)
         rates.append(sampling_rate)
 
+    # TODO: Clean up this code block above....
+    # Quick fix to make plots prettier
```
Comment (ClFeSc) [inline pysrc/models/metanome_run.py]: Do you want to keep this and the things below like this, or do you want to change it back before merging?

File: pysrc/scripts/evaluation.py [hunk 5]
```diff
@@ -35,7 +35,7 @@ def create_evaluation_csv(runs: MetanomeRunBatch, output_file: str, output_folde
 
     with open(output_csv, 'w') as csv_output:
         writer = csv.writer(csv_output, quoting=csv.QUOTE_ALL)
-        writer.writerow(['sampled_files', 'sampling_method', "sampling_rate", 'tp', 'fp', 'fn', 'precision', 'recall', 'f1'])
+        writer.writerow(['sampling_method', "sampling_rate", 'tp', 'fp', 'fn', 'precision', 'recall', 'f1'])
```
Comment (ClFeSc) [inline pysrc/scripts/evaluation.py]: What about the changes in this file?

**Review by ClFeSc — COMMENTED** [review]
My first opinion. I'll try it out later.

File: pysrc/models/metanome_run.py (hunk 4 above)
Comment (dkuska) [reply pysrc/models/metanome_run.py]: That really depends on what we want to do in the future with this.

Looking forward are we going to be mixing sampling methods or just having a single one in each Run?

I'd say lets talk about this on tuesday.

File: pysrc/scripts/evaluation.py (hunk 5 above)
Comment (dkuska) [reply pysrc/scripts/evaluation.py]: I felt that we never really care about the specific file names and as such there is no need to keep them in the output csv

File: pysrc/models/ind.py (hunk 1 above)
Comment (ClFeSc) [reply pysrc/models/ind.py]: @dkuska I think I've found a solution: You can use [`dataclasses-json`](https://pypi.org/project/dataclasses-json/) to parse and write JSON. Just add `@dataclass_json` to each `@dataclass`, and replace `json` calls with the methods from `dataclasses-json`.

**Comment by yjojo17** [comment]
I'll test it tomorrow in BDS maybe to get this PR through would be nice so i can generate plots for the new sampling methods

**Comment by dkuska** [comment]
Alright, so loading the MetanomeRunBatch from the json works now.
However something in the branch is fucky with the new additions of the run-folders by Clemens. I dont really get why.
So this means, that we will need to rework the evaluations anyway, however I'd just like to merge this and give you the opportunity to work on other things...

**Comment by ClFeSc** [comment]
@dkuska Are you sure the code is still working? I get an error when calling `run_metanome`:

```plaintext
  File "pysrc/scripts/sampling.py", line 246, in run_experiments
    result = run_metanome(configuration, output_file_name, config.pipe)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: run_metanome() takes 2 positional arguments but 3 were given
```

File: pysrc/models/ind.py [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
+@dataclass_json
 @dataclass(frozen=True)
 class IND:
     """Represents a single, nary ind"""
     dependents: list[ColumnInformation]
 
     referenced: list[ColumnInformation]
 
-    errors: list[ErrorMetric] = field(default_factory=list, compare=False)
+    errors: list[ErrorMetric] = field(default_factory=list, compare=False) #NOTE: THIS IS JUST TEMPORARY SO THAT WE CAN ACTUALLY WORK WITH PARTIAL SPIDER!!!
```
Comment (ClFeSc) [inline pysrc/models/ind.py]: Remove comment.

File: pysrc/models/metanome_run.py [hunk 7]
```diff
@@ -235,16 +254,23 @@ def parse_results(result_file_name: str, arity: str, results_folder: str, print_
     return MetanomeRunResults(ind_list)
 
 
-def run_metanome(configuration: MetanomeRunConfiguration, output_fname: str, pipe: bool) -> MetanomeRun:
-    # Make these configurable
+def run_metanome(configuration: MetanomeRunConfiguration, output_fname: str) -> MetanomeRun:
```
Comment (ClFeSc) [inline pysrc/models/metanome_run.py]: Re-add `pipe` parameter.

File: requirements.txt [hunk 8]
```diff
@@ -6,3 +6,4 @@ seaborn~=0.12.1
 python-dateutil~=2.8.2
 numpy~=1.23.4
 matplotlib~=3.6.2
+dataclasses-json
```
Comment (ClFeSc) [inline requirements.txt]: Remove `dacite` (at the top) and add a version like the other dependencies.

**Review by ClFeSc — COMMENTED** [review]
For now my code review.

**Comment by dkuska** [comment]
> For now my code review.

All integrated :+1:

File: pysrc/models/metanome_run.py [hunk 9]
```diff
@@ -33,10 +36,13 @@ class MetanomeRunConfiguration:
     is_baseline: bool
 
     def __hash__(self) -> int:
-        return hash((self.arity, tuple(self.sampling_rates), tuple(self.sampling_methods), self.time, self.source_dir, tuple(self.source_files), self.tmp_folder, self.results_folder,
-        self.result_suffix, self.output_folder, self.clip_output, self.header, self.print_inds, self.create_plots, self.is_baseline))
+        return hash((self.algorithm, self.arity, tuple(self.sampling_rates), tuple(self.sampling_methods), self.time, self.source_dir,
+                     tuple(self.source_files), self.tmp_folder, self.results_folder,
+                     self.result_suffix, self.output_folder, self.output_file, self.clip_output, self.header,
```
Comment (ClFeSc) [inline pysrc/models/metanome_run.py]:
```suggestion
                     self.result_suffix, self.output_folder, self.clip_output, self.header,
```

The property doesn't exist anymore.

File: pysrc/models/metanome_run.py [hunk 10]
```diff
@@ -151,6 +162,7 @@ def parse_results(result_file_name: str, arity: str, results_folder: str, print_
 
     for line in lines:
         line_json = json.loads(line)
+        errors: list[MissingValues] = []
```
Comment (ClFeSc) [inline pysrc/models/metanome_run.py]:
```suggestion
        errors: list[ErrorMetric] = []
```

File: requirements.txt [hunk 11]
```diff
@@ -6,3 +6,4 @@ seaborn~=0.12.1
 python-dateutil~=2.8.2
 numpy~=1.23.4
 matplotlib~=3.6.2
+dataclasses-json~=0.5.7
```
Comment (ClFeSc) [inline requirements.txt:8]: Still remove `dacite`. We don't need it anymore, I think.

**Review by ClFeSc — COMMENTED** [review]
A few more comments...

**Review by yjojo17 — APPROVED** [review]

**Review by ClFeSc — APPROVED** [review]
Let's go ahead and merge this.

**Pull request merged**